### PR TITLE
Defines CODEOWNERS for covid_vaccine module

### PR DESCRIPTION
**Pull request by patrickvinograd**
## Description of change
Defines CODEOWNERS for covid_vaccine module and spec directories, assigning to cto-health-care-products team. This aligns with ownership of the corresponding frontend app.

## Original issue(s)
department-of-veterans-affairs/va.gov-team#0000

### CODE REVIEW

**Review by thilton-oddball — APPROVED** [review]
LGTM

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -23,3 +23,8 @@ app/models/user.rb @department-of-veterans-affairs/vsp-identity
 app/models/user_identity.rb @department-of-veterans-affairs/vsp-identity
 app/models/user_session_form.rb @department-of-veterans-affairs/vsp-identity
 lib/saml @department-of-veterans-affairs/vsp-identity
+
+# covid_vaccine product
+modules/covid_vaccine @department-of-veterans-affairs/va-cto-health-products
```
Comment (erluti) [inline .github/CODEOWNERS:28]:
```suggestion
modules/covid_vaccine @department-of-veterans-affairs/va-cto-health-products @department-of-veterans-affairs/backend-review-group
```

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -23,3 +23,8 @@ app/models/user.rb @department-of-veterans-affairs/vsp-identity
...
+spec/support/*/covid_vaccine @department-of-veterans-affairs/va-cto-health-products
```
Comment (erluti) [inline .github/CODEOWNERS:29]:
```suggestion
spec/support/*/covid_vaccine @department-of-veterans-affairs/va-cto-health-products @department-of-veterans-affairs/backend-review-group
```

**Review by erluti — COMMENTED** [review]
Can we also make the backend-review-group co-owners so app-wide maintenance stuff doesn't get blocked?

**Pull request merged**